src/Verify/VirtualizedRunHelper.cs

```diff
@@ -0,0 +1,115 @@
+class VirtualizedRunHelper
+{
+    // e.g. WSL or docker run (https://github.com/VerifyTests/Verify#unit-testing-inside-virtualized-environment)
+    readonly bool appearsToBeLocalVirtualizedRun;
+    readonly string originalCodeBaseRootAbsolute = string.Empty;
+    readonly string mappedCodeBaseRootAbsolute = string.Empty;
+    static readonly char[] Separators = { '\\', '/' };
+
+    public VirtualizedRunHelper(Assembly userAssembly)
+    {
+        var originalCodeBaseRoot = AttributeReader.TryGetSolutionDirectory(userAssembly, false, out var solutionDir)
+            ? solutionDir
+            : AttributeReader.GetProjectDirectory(userAssembly);
+        var appearsToBeBuiltOnDifferentPlatform =
+            !string.IsNullOrEmpty(originalCodeBaseRoot) &&
+            !originalCodeBaseRoot.Contains(Path.DirectorySeparatorChar) &&
+            originalCodeBaseRoot.Contains("\\");
+
+        if (!appearsToBeBuiltOnDifferentPlatform)
+        {
+            return;
+        }
+
+        string currentDir = Environment.CurrentDirectory;
+        string currentDirRelativeToAppRoot = currentDir.TrimStart(Separators);
+
+        // WSL paths mount to /mnt/<drive>/...
+        // docker testing mounts to /mnt/approot/...
+        if (!TryRemoveDirFromStartOfPath(ref currentDirRelativeToAppRoot) ||
+            !TryRemoveDirFromStartOfPath(ref currentDirRelativeToAppRoot))
```

Comment:
why is this line duplicated?

Reply:
ahh i see. each time it modifies the ref param.

Reply:
This could have surely been better commented :-|